datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -303,7 +304,23 @@ def _job_loop(self):
                         "dd.{}.async_job.inactive_stop".format(self._dbms), 1, tags=self._job_tags, raw=True
                     )
                     break
-                self._run_job_rate_limited()
+                if 'profile_memory' in self._check.init_config or (
+                    datadog_agent.tracemalloc_enabled() and should_profile_memory(datadog_agent, self._check.name)
+                ):
+                    from datadog_checks.base.utils.agent.memory import profile_memory
+
+                    metrics = profile_memory(
+                        self._run_job_rate_limited,
+                        self._check.init_config,
+                        namespaces=[self._check.name, self._job_name],
+                    )
+
+                    tags = self._check.get_debug_metric_tags() + self._job_tags
+                    tags.extend(self._check.instance.get('__memory_profiling_tags', []))
+                    for m in metrics:
+                        self._check.gauge(m.name, m.value, tags=tags, raw=True)
+                else:
+                    self._run_job_rate_limited()
```

Comment:
Instead of copy/pasting the code from the `AgentCheck`, I would rather add a function that is called in both places